### PR TITLE
DB Migration + API for Geospatial queries

### PR DESCRIPTION
**Pull request by KnightNiwrem**

### CODE REVIEW

**Comment by KnightNiwrem** [comment]
Dependent PR: https://github.com/nus-chatbox/quack/pull/8

File: api-server/server.js [hunk 1]
```diff
@@ -124,6 +140,42 @@ app.post("/authenticate", (req, res) => {
...
 
+
+/******************************* Rooms ********************************/
+
+app.get("/rooms", passport.authenticate(["jwt"], { session: false }), (req, res) => {
+  let userLatitude = req.user.latitude;
+  let userLongitude = req.user.longitude;
+
+  let nearbyRoomPromise = Room.query().select(
+    raw("*, (6371 * acos(cos(radians(:userLatitude)) * " +
```
Comment (ngsikai) [inline api-server/server.js:151]: dat trigo tho. 6371 magic number? and `acos(cos(` is to get degrees ah. My trigo knowledge has floated away for quite a while lol.

File: api-server/server.js [hunk 2]
```diff
@@ -124,6 +140,42 @@ app.post("/authenticate", (req, res) => {
...
+  })).having("distance", "<=", 1).orderBy("distance", "asc");
+
+  nearbyRoomPromise.then((rooms) => {
+    res.json({
+      rooms: rooms
+    });
+  });
+});
+
+app.get("/subscriptions", passport.authenticate(["jwt"], { session: false }), (req, res) => {
```
Comment (ngsikai) [inline api-server/server.js:168]: subscriptions are rooms that a users are in ah?

File: src/store/login/index.js [hunk 3]
```diff
@@ -4,8 +4,8 @@
 /* eslint-disable prefer-template */
 export default {
   state: {
-    fbToken: null,
-    jwtToken: null
+    fbToken: window.localStorage.getItem('fbToken'),
```
Comment (ngsikai) [inline src/store/login/index.js:7]: noice

**Review by ngsikai — APPROVED** [review]

File: api-server/server.js (hunk 1 above)
Comment (KnightNiwrem) [reply api-server/server.js:151]: Earth radius in kilometres

File: api-server/server.js (hunk 2 above)
Comment (KnightNiwrem) [reply api-server/server.js:168]: Yup. Might delete it with migrations later. See first. Lol.

**Pull request merged**